src/analysis_and_optimization/Pedantic_analysis.ml

```diff
@@ -261,16 +263,14 @@ let compiletime_value_of_expr
 let list_distributions (mir : Program.Typed.t) : dist_info Set.Poly.t =
   let take_dist (expr : Expr.Typed.t) =
     match expr.pattern with
-    | Expr.Fixed.Pattern.FunApp (StanLib fname, arg_exprs) -> (
-      match chop_dist_name fname with
-      | Some dname ->
-          let params = parameter_set mir in
-          let data = data_set mir in
-          let args =
-            List.map ~f:(compiletime_value_of_expr params data) arg_exprs
-          in
-          Some {name= dname; loc= expr.meta.loc; args}
-      | _ -> None )
+    | Expr.Fixed.Pattern.FunApp (StanLib (fname, FnLpdf true), arg_exprs) ->
+        let fname = chop_dist_name fname |> Option.value_exn in
```

Comment:
It's interesting that we use value_exn now because the representation is redundant. I wonder if we should cut the suffix off of the string when we parse the `suffix` type? What do you think?

Reply:
I guess that would probably be too complicated to translate back and forth for e.g. codegen and signature lookup. Hmm